src/libvirtApi/common.js

```diff
@@ -161,88 +161,91 @@ function doUsagePolling(name, connectionName, objPath) {
+async function getNodeMaxMemory({ connectionName }) {
     // Some nodes don"t return all memory in just one cell.
     // Using -1 == VIR_NODE_MEMORY_STATS_ALL_CELLS will return memory across all cells
-    return call(connectionName, "/org/libvirt/QEMU", "org.libvirt.Connect", "NodeGetMemoryStats", [-1, 0], { timeout, type: "iu" })
-            .then(stats => store.dispatch(setNodeMaxMemory({ memory: stats[0].total })))
-            .catch(ex => {
-                console.warn("NodeGetMemoryStats failed:", ex.toString());
-                return Promise.reject(ex);
-            });
+    try {
+        const [stats] = await call(connectionName, "/org/libvirt/QEMU", "org.libvirt.Connect", "NodeGetMemoryStats", [-1, 0], { timeout, type: "iu" });
+        store.dispatch(setNodeMaxMemory({ memory: stats.total }));
+    } catch (ex) {
+        console.warn("NodeGetMemoryStats failed:", ex.toString());
+        throw ex;
+    }
 }
 
-function getOsInfoList () {
+async function getOsInfoList () {
     logDebug(`GET_OS_INFO_LIST():`);
-    return python.spawn(getOSListScript, null, { err: "message", environ: ['LC_ALL=C.UTF-8'] })
-            .then(osList => {
-                parseOsInfoList(osList);
-            })
-            .catch(ex => {
-                console.error(`get os list returned error: "${JSON.stringify(ex)}"`);
-                parseOsInfoList('[]');
-            });
+    try {
+        const osList = await python.spawn(getOSListScript, null, { err: "message", environ: ['LC_ALL=C.UTF-8'] });
+        parseOsInfoList(osList);
+    } catch (ex) {
+        console.error(`get os list returned error: "${JSON.stringify(ex)}"`);
+        parseOsInfoList('[]');
+    }
 }
 
-function networkUpdateOrDelete(connectionName, netPath) {
-    call(connectionName, "/org/libvirt/QEMU", "org.libvirt.Connect", "ListNetworks", [0], { timeout, type: "u" })
-            .then(objPaths => {
-                if (objPaths[0].includes(netPath))
-                    return networkGet({ connectionName, id: netPath, updateOnly: true });
-                else // Transient network which got undefined when stopped
-                    return store.dispatch(undefineNetwork({ connectionName, id: netPath }));
-            })
-            .catch(ex => console.warn("networkUpdateOrDelete action failed:", ex.toString()));
+async function networkUpdateOrDelete(connectionName, netPath) {
+    try {
+        const [objPaths] = await call(connectionName, "/org/libvirt/QEMU", "org.libvirt.Connect", "ListNetworks", [0],
+                                      { timeout, type: "u" });
+        if (objPaths.includes(netPath))
+            return networkGet({ connectionName, id: netPath, updateOnly: true });
+        else // Transient network which got undefined when stopped
+            store.dispatch(undefineNetwork({ connectionName, id: netPath }));
+    } catch (ex) {
+        console.warn("networkUpdateOrDelete action failed:", ex.toString());
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1400-20240125-085010-77feb7f2-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)